### PR TITLE
front: fix infra lock status

### LINKED ISSUE #6952: When infra is locked, front displays that we can still click on save
### What happened?

Capture d’écran 2024-03-20 à 16 37 41

### What did you expect to happen?

Save button should be locked, a message should be displayed

### How can we reproduce it (as minimally and precisely as possible)?

go to infra editor, try to save a new object, save button is available 

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Nixos v22.11
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

dev 5c866c3

### PR DESCRIPTION
**Pull request by Akctarus**
closes #6952

### CODE REVIEW

File: front/src/applications/editor/tools/rangeEdition/tool-factory.tsx [hunk 1]
```diff
@@ -97,10 +97,10 @@ function getRangeEditionTool<T extends EditorRange>({
           id: `save-${objectTypeAction}`,
           icon: AiFillSave,
           labelTranslationKey: `Editor.tools.${objectTypeEdition}-edition.actions.save-${objectTypeAction}`,
-          isDisabled({ isLoading, state }) {
+          isDisabled({ isLoading, isInfraLocked, state }) {
             if (isLoading) return true;
             if (canSave) return !canSave(state);
-            return false;
+            return isInfraLocked || false;
```
Comment (Uriel-Sautron) [inline front/src/applications/editor/tools/rangeEdition/tool-factory.tsx]:
```suggestion
            return !!isInfraLocked;
```

**Review by Uriel-Sautron — COMMENTED** [review]
LGTM, good work thank you.
Just a comment.

File: front/src/applications/editor/tools/rangeEdition/tool-factory.tsx (hunk 1 above)
Comment (Akctarus) [reply front/src/applications/editor/tools/rangeEdition/tool-factory.tsx]: Thank you ! Fixed

File: front/src/applications/editor/Editor.tsx [hunk 2]
```diff
@@ -65,6 +65,8 @@ const Editor = () => {
   }, [setRenderingFingerprint]);
 
   const [isFormSubmited, setIsFormSubmited] = useState(false);
+  const [getInfraById, { data: infra }] = osrdEditoastApi.endpoints.getInfraById.useLazyQuery({});
```
Comment (clarani) [inline front/src/applications/editor/Editor.tsx]: Don't use a lazy query, use useQuery directly

**Review by clarani — CHANGES_REQUESTED** [review]
The save button is not disabled in the speed limit tab :/ 
image

File: front/src/applications/editor/Editor.tsx (hunk 2 above)
Comment (Akctarus) [reply front/src/applications/editor/Editor.tsx]: Fixed !

**Review by clarani — APPROVED** [review]
LGTM, tested :white_check_mark:

**Pull request merged**